applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_structural_response_function.h

```diff
@@ -0,0 +1,1249 @@
+            {
+                const Variable<double>& r_variable =
+                    KratosComponents<Variable<double>>::Get(label);
+
+                //check for output variable to save later the computed sensitivity
+                if( KratosComponents<Variable<double>>::Has(output_label) == true )
+                {
+                    const Variable<double>& r_output_variable =
+                        KratosComponents<Variable<double>>::Get(output_label);
+                    this->UpdateConditionSensitivities(r_variable, r_output_variable);
+                }
+                else
+                    KRATOS_ERROR << "Unsupported condition variable for output: " << output_label << "." << std::endl;
+            }
+            else if (KratosComponents<Variable<array_1d<double,3>>>::Has(label) == true)
+            {
+                const Variable<array_1d<double,3>>& r_variable =
+                    KratosComponents<Variable<array_1d<double,3>>>::Get(label);
+
+                //check for output variable to save later the computed sensitivity
+                if (KratosComponents<Variable<array_1d<double,3>>>::Has(output_label) == true)
+                {
+                    const Variable<array_1d<double,3>>& r_output_variable =
+                        KratosComponents<Variable<array_1d<double,3>>>::Get(output_label);
+                    this->UpdateConditionSensitivities(r_variable, r_output_variable);
+                }
+                else
+                    KRATOS_ERROR << "Unsupported condition variable for output: " << output_label << "." << std::endl;
+
+            }
+            else
+                KRATOS_ERROR << "Unsupported condition variable: " << label << "." << std::endl;
+        }
+
+        KRATOS_CATCH("");
+    }
+
+
+    /// Calculate the scalar valued response function
+    virtual double CalculateValue(ModelPart& rModelPart)
+    {
+        return 0.0;
+    }
+
+    ///@}
+
+protected:
+    ///@name Protected member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /// Calculate and add the sensitivity from the current adjoint and primal solutions.
+    /*
+     * This function updates the accumulated sensitivities for a single time step.
+     * The accumulated sensitivity is defined as:
+     *
+     * \f[
+     * d_{\mathbf{s}}\bar{J} = \Sigma_{n=1}^N
+     *   (\partial_{\mathbf{s}}J^n + \lambda^{nT}\partial_{\mathbf{s}}\mathbf{r}^n)
+     *    \Delta t
+     * \f]
+     *
+     * with \f$\mathbf{r}^n\f$ the residual of the governing partial differential
+     * equation for the current step. This function should be called once for
+     * each design variable per time step.
+     */
+   /* template <typename TDataType>
+    void UpdateNodalSensitivities(Variable<TDataType> const& rSensitivityVariable)
```

Comment:
whats the difference between this and the not commented function below?

Reply:
It is not needed. I removed it.